algorithms/indexing/indexer.py

```diff
@@ -705,7 +705,18 @@ def index(self):
                 rotation_matrix_differences(self.refined_experiments.crystals())
             )
 
-        self._xyzcal_mm_to_px(self.experiments, self.refined_reflections)
+        # in case refinement failed, clean the reflections which may have had
+        # indices assigned to an experiment which was not saved...
+        refl = self.refined_reflections
+        expt = self.refined_experiments
+        spare = set(refl["id"]) - set(range(len(expt))) - set([-1])
+        for s in spare:
+            sel = refl["id"] == s
+            logger.info("Removing %d reflections with id %d" % (sel.count(True), s))
+            refl["id"].set_selected(sel, -1)
+            refl.unset_flags(sel, refl.flags.indexed)
```

Comment:
This code would probably be better as part of the try/except where `self.refine()` is called, i.e. when the failed refinement was actually trapped.

Reply:
👍

Agreed working on this now